src/background/certificate/providers/InBrowserProvider.ts

```diff
@@ -15,17 +16,31 @@ export class InBrowserProvider implements CertificateProvider {
 
   async getCertificate(
     requestDetails: WebRequest.OnHeadersReceivedDetailsType
-  ): Promise<RawCertificate> {
-    const { requestId } = requestDetails;
-    const securityInfo = await this.getSecurityInfo(requestId, {
-      certificateChain: false,
-      rawDER: true,
-    });
+  ): Promise<RawCertificateResponse> {
+    return new Promise(async (resolve, reject) => {

```

Comment:
I think its unnecessary to explicitly return a promise here, since by using the async/await syntax a promise is returned anyway.

Reply:
And then I should throw a RawCertificateResponse? Looks weird to me.

Reply:
No I can't since the promise has to reject.